roofit/hs3/src/RYMLParser.cxx

```diff
@@ -0,0 +1,272 @@
+  // write an float to this node    
+  node->get() << d;
+  return *this;
+}
+
+const TRYMLTree::Node& TRYMLTree::Node::operator>> (std::string &v) const {
+  // read a string from this node
+  node->get() >> v;
+  return *this;
+}
+
+TRYMLTree::Node& TRYMLTree::Node::operator[] (std::string const& k) {
+  // get a child node with the given key
+  return Impl::mkNode(tree,node->get()[c4::to_csubstr(tree->incache(k))]);
+}
+
+TRYMLTree::Node& TRYMLTree::Node::operator[] (size_t pos) {
+  // get a child node with the given index
+  return Impl::mkNode(tree,node->get()[pos]);     
+}
+
+const TRYMLTree::Node& TRYMLTree::Node::operator[] (std::string const& k) const {
+  // get a child node with the given key (const version)  
+  return Impl::mkNode(tree,node->get()[c4::to_csubstr(tree->incache(k))]);     
+}
+
+const TRYMLTree::Node& TRYMLTree::Node::operator[] (size_t pos) const {
+  // get a child node with the given index (const version)    
+  return Impl::mkNode(tree,node->get()[pos]);          
+}
+
+bool TRYMLTree::Node::is_container() const {
+  // return true if this node can have child nodes
+  return node->get().is_container();
+}
+
+bool TRYMLTree::Node::is_map() const {
+  // return true if this node is a map (JSON object)
+  return node->get().is_map();
+}
+
+bool TRYMLTree::Node::is_seq() const {
+  // return true if this node is a sequence (JSON array)  
+  return node->get().is_seq();
+}
+
+std::string TRYMLTree::Node::key() const {
+  // obtain the key of this node
+  std::stringstream ss;    
+  ss << node->get().key();
+  return ss.str();
+}
+
+std::string TRYMLTree::Node::val() const {;
+  // obtain the value of this node (as a string)
+  std::stringstream ss;    
+  ss << node->get().val();
+  return ss.str();
+}
+
+TRYMLTree::Node& TRYMLTree::Node::append_child(){
+  // append a new child to this node
+  return Impl::mkNode(tree,node->get().append_child());
+}
+
+bool TRYMLTree::Node::has_key() const {
+  // return true if this node has a key
+  return node->get().has_key();
+}
+
+bool TRYMLTree::Node::has_val() const {
+  // return true if this node has a value  
+  return node->get().has_val();
+}
+
+bool TRYMLTree::Node::has_child(std::string const&s) const {
+  // return true if this node has a child with the given key
+  return node->get().has_child(c4::to_csubstr(s.c_str()));
+}
+
+size_t TRYMLTree::Node::num_children() const {
+  // return the number of child nodes of this node
+  return node->get().num_children();
+}
+
+TRYMLTree::Node& TRYMLTree::Node::child(size_t pos){
+  // return the child with the given index
+  return Impl::mkNode(tree,node->get().child(pos));
+}
+
+const TRYMLTree::Node& TRYMLTree::Node::child(size_t pos) const {
+  // return the child with the given index (const version)
+  return Impl::mkNode(tree,node->get().child(pos));
+}
+
+// specific functions
+
+namespace {
+  void error_cb(const char* msg, size_t msg_len, void* /*user_data*/){
```

Comment:
```suggestion
  void error_cb(const char* msg, size_t msg_len, void* /*user_data*/){
  void error_cb(const char* msg, size_t msg_len, c4::yml::Location /*not used*/, void* /*user_data*/){
```

This doesn't match the signature expected by the newest rapidyaml (I tried to compile with the `master` branch from https://github.com/biojppm/rapidyaml). Can you just update the signature? Or do you need backwards compatibility with older rapidyaml? I don't think we need that though, since the RooFit HS3 is a new package anyway.